monero-rpc/src/monerod.rs

```diff
@@ -1,46 +1,91 @@
 use anyhow::{Context, Result};
-use serde::Deserialize;
+use monero::{cryptonote::hash::Hash, util::ringct, PublicKey};
+use serde::{de::DeserializeOwned, Deserialize, Serialize, Serializer};
 
 #[jsonrpc_client::api(version = "2.0")]
 pub trait MonerodRpc {
     async fn generateblocks(&self, amount_of_blocks: u32, wallet_address: String)
         -> GenerateBlocks;
     async fn get_block_header_by_height(&self, height: u32) -> BlockHeader;
     async fn get_block_count(&self) -> BlockCount;
+    async fn get_block(&self, height: u32) -> GetBlockResponse;
 }
 
 #[jsonrpc_client::implement(MonerodRpc)]
 #[derive(Debug, Clone)]
 pub struct Client {
     inner: reqwest::Client,
     base_url: reqwest::Url,
+    get_o_indexes_bin_url: reqwest::Url,
+    get_outs_bin_url: reqwest::Url,
 }
 
 impl Client {
     /// New local host monerod RPC client.
     pub fn localhost(port: u16) -> Result<Self> {
+        Self::new("127.0.0.1".to_owned(), port)
+    }
+
+    fn new(host: String, port: u16) -> Result<Self> {
         Ok(Self {
             inner: reqwest::ClientBuilder::new()
                 .connection_verbose(true)
                 .build()?,
-            base_url: format!("http://127.0.0.1:{}/json_rpc", port)
+            base_url: format!("http://{}:{}/json_rpc", host, port)
                 .parse()
                 .context("url is well formed")?,
+            get_o_indexes_bin_url: format!("http://{}:{}/get_o_indexes.bin", host, port)
```

Comment:
:open_mouth: So it's a different base URL for these particular calls.

Reply:
Yes, they are in fact "normal" HTTP calls (except that they don't care about the method you pass in) with a binary payload.